app/src/main/java/org/kiwix/kiwixmobile/webserver/WebServerHelper.java

```diff
@@ -0,0 +1,66 @@
+package org.kiwix.kiwixmobile.webserver;
+
+import android.util.Log;
+import androidx.annotation.NonNull;
+import java.util.ArrayList;
+import javax.inject.Inject;
+import org.kiwix.kiwixlib.JNIKiwixException;
+import org.kiwix.kiwixlib.JNIKiwixLibrary;
+import org.kiwix.kiwixlib.JNIKiwixServer;
+import org.kiwix.kiwixmobile.utils.ServerUtils;
+
+/**
+ * WebServerHelper class is used to set up the suitable environment i.e. getting the
+ * ip address and port no. before starting the WebServer
+ * Created by Adeel Zafar on 18/07/2019.
+ */
+
+public class WebServerHelper {
+  private static final String TAG = "WebServerHelper";
+  private JNIKiwixLibrary kiwixLibrary;
+  private JNIKiwixServer kiwixServer;
+
+  @Inject public WebServerHelper(@NonNull JNIKiwixLibrary kiwixLibrary,
+      @NonNull JNIKiwixServer kiwixServer) {
+    this.kiwixLibrary = kiwixLibrary;
+    this.kiwixServer = kiwixServer;
+  }
+
+  public boolean startServerHelper(@NonNull ArrayList<String> selectedBooksPath) {
+
+    String ip = ServerUtils.getIpAddress();
+    ip = ip.replaceAll("\n", "");
+    if (ip.length() == 0) {
+      return false;
+    } else if (startAndroidWebServer(selectedBooksPath)) {
+      return true;
+    }
+    return ServerUtils.isServerStarted;
+  }
+
+  public void stopAndroidWebServer() {
+    if (ServerUtils.isServerStarted) {
```

Comment:
this is the class that knows if the server is started, it doesn't need to ask ServerUtils

Reply:
@iadeelzafar can we introduce a field here that keeps track of if the server is started so all this class does is update the static variable

Reply:
Fixed 7efd39e3ee276d5a125233c1da25a3e77afd9665